common/src/main/java/com/microsoft/identity/common/internal/authorities/AzureActiveDirectoryB2CAuthority.java

```diff
@@ -93,4 +93,21 @@ public OAuth2Strategy createOAuth2Strategy(@NonNull final OAuth2StrategyParamete
         MicrosoftStsOAuth2Configuration config = createOAuth2Configuration();
         return new MicrosoftStsOAuth2Strategy(config, parameters);
     }
+
+    /**
+     * This method attempts to split the Authority URI
+     * and return the last item, which is the policy name.
+     * The authority format for Azure AD B2C is: https://{azureADB2CHostname}/tfp/{tenant}/{policyName}
+     *
+     * @return a String with the Policy name
+     */
+    public String getB2CPolicyName(){
+        try {
+            final String authorityUriString = this.getAuthorityUri().toString();
+            final String [] authorityUriParts = authorityUriString.split("/");
+            return authorityUriParts[authorityUriParts.length - 1];
+        } catch (Exception e) {
```

Comment:
Can we catch a more specific exception here?

Reply:
43f69c5bd5c9dbb87f235509c77ac18aa1ed42a9